gensim/models/doc2vec.py

```diff
@@ -808,6 +809,50 @@ def delete_temporary_training_data(self, keep_doctags_vectors=True, keep_inferen
         if self.docvecs and hasattr(self.docvecs, 'doctag_syn0_lockf'):
             del self.docvecs.doctag_syn0_lockf
 
+    def save_word2vec_format(self, fname, doc_vec=True, word_vec=False, prefix='dt_', fvocab=None, binary=False):
+        """
+        Store the input-hidden weight matrix.
+
+         `fname` is the file used to save the vectors in
+         `doc_vec` is an optional boolean indicating whether to store document vectors
+         `word_vec` is an optional boolean indicating whether to store word vectors
+         (if both doc_vec and word_vec are True, then both vectors are stored in the same file)
+         `prefix` to uniquely indentify doctags from word vocab, and avoid collision
+         in case of repeated string in doctag and word vocab
+         `fvocab` is an optional file used to save the vocabulary
+         `binary` is an optional boolean indicating whether the data is to be saved
+         in binary word2vec format (default: False)
+
+        """
+        total_vec = len(self.docvecs)
+        if word_vec:
+            total_vec = total_vec + len(self.wv.vocab)
+        # save document vectors
+        if doc_vec:
+            logger.info("storing %sx%s projection weights into %s" % (total_vec, self.vector_size, fname))
+            with utils.smart_open(fname, 'wb') as fout:
```

Comment:
If all writing is done in this method, seems that should be possible, and would be cleaner, to only open the file once, rather once then a second time in append-mode. BUT, see later comment for a way that this method might be able to offload some of the writing, and thus would just choose to re-open append.